lib/api/node.go

```diff
@@ -63,11 +63,19 @@ type NodeBGPSpec struct {
 
 	// IPv4Address is the IPv4 address of this node.  At least one of the
 	// IPv4 and IPv6 addresses should be specified.
-	IPv4Address *net.IP `json:"ipv4Address,omitempty" validate:"omitempty,ipv4"`
+	IPv4Address *net.IP `json:"ipv4Address,omitempty" validate:"omitempty"`
+
+	// IPv4Network is the IPv4 local network that this node is on (corresponding
+	// to the configured IPv4 address).
+	IPv4Network *net.IPNet `json:"ipv4Network,omitempty" validate:"omitempty"`
```

Comment:
It also feels weird to put the IPv4Network field in the BGP config for a node, since it's not actually used for BGP from a users perspective - it's used for ipip tunneling.

Maybe these should be moved to:

```
Node.Spec.IPv4Network
Node.Spec.IPv6Network
```

Then again, the presence of the BGP section does sort of dictate whether or not these fields will be respected :(